File: csharp-package/brainflow/brainflow/brainflow_input_params.cs
using System.Collections.Generic;
using System.IO;
using System.Runtime.Serialization;
using System.Runtime.Serialization.Json;
using System.Text;


namespace brainflow
{
    public enum IpProtocolType
    {
        NONE = 0,
        UDP = 1,
        TCP = 2
    };

    /// <summary>
    ///  Check SuportedBoards to get information about fields which are required for specific board
    /// </summary>
    [DataContract]
    public class BrainFlowInputParams
    {
        /// <summary>
        /// serial port name
        /// </summary>
        [DataMember]
        public string serial_port;
        /// <summary>
        /// MAC address
        /// </summary>
        [DataMember]
        public string mac_address;
        /// <summary>
        /// IP address
        /// </summary>
        [DataMember]
        public string ip_address;
        /// <summary>
        /// PORT
        /// </summary>
        [DataMember]
        public int ip_port;
        /// <summary>
        /// IP protocol, use IpProtocolType
        /// </summary>
        [DataMember]
        public int ip_protocol;
        /// <summary>
        /// you can provide additional info to low level API using this field
        /// </summary>
        [DataMember]
        public string other_info;
        /// <summary>
        /// timeout for device discovery or connection
        /// </summary>
        [DataMember]
        public int timeout;
        /// <summary>
        /// serial number
        /// </summary>
        [DataMember]
        public string serial_number;
        /// <summary>
        /// file
        /// </summary>
        [DataMember]
        public string file;

        public BrainFlowInputParams ()
        {
            serial_port = "";
            mac_address = "";
            ip_address = "";
            ip_port = 0;
            ip_protocol = (int)IpProtocolType.NONE;
            other_info = "";
            timeout = 0;
            serial_number = "";
            file = "";
        }

        public string to_json ()
        {
            // copypaste from https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/how-to-serialize-and-deserialize-json-data?redirectedfrom=MSDN
            var ms = new MemoryStream ();
            var ser = new DataContractJsonSerializer (typeof (BrainFlowInputParams));
            ser.WriteObject (ms, this);
            byte[] json = ms.ToArray ();
            ms.Close ();
            return Encoding.UTF8.GetString (json, 0, json.Length);
        }
    }
}

File: java-package/brainflow/src/main/java/brainflow/BrainFlowInputParams.java
package brainflow;

import com.google.gson.Gson;

/**
 * to get fields which are required for your board check SupportedBoards section
 */
public class BrainFlowInputParams
{
    public String ip_address;
    public String mac_address;
    public String serial_port;
    public int ip_port;
    public int ip_protocol;
    public String other_info;
    public int timeout;
    public String serial_number;
    public String file;

    public BrainFlowInputParams ()
    {
        ip_address = "";
        mac_address = "";
        serial_port = "";
        ip_port = 0;
        ip_protocol = IpProtocolType.NONE.get_code ();
        other_info = "";
        timeout = 0;
        serial_number = "";
        file = "";
    }

    public String to_json ()
    {
        return new Gson ().toJson (this);
    }

    public String get_ip_address ()
    {
        return ip_address;
    }

    public void set_ip_address (String ip_address)
    {
        this.ip_address = ip_address;
    }

    public String get_mac_address ()
    {
        return mac_address;
    }

    public void set_mac_address (String mac_address)
    {
        this.mac_address = mac_address;
    }

    public String get_serial_port ()
    {
        return serial_port;
    }

    public void set_serial_port (String serial_port)
    {
        this.serial_port = serial_port;
    }

    public int get_ip_port ()
    {
        return ip_port;
    }

    public void set_ip_port (int ip_port)
    {
        this.ip_port = ip_port;
    }

    public int get_ip_protocol ()
    {
        return ip_protocol;
    }

    public void set_ip_protocol (int ip_protocol)
    {
        this.ip_protocol = ip_protocol;
    }

    public String get_other_info ()
    {
        return other_info;
    }

    public void set_other_info (String other_info)
    {
        this.other_info = other_info;
    }

    public void set_timeout (int timeout)
    {
        this.timeout = timeout;
    }

    public int get_timeout ()
    {
        return timeout;
    }

    public String get_serial_number ()
    {
        return serial_number;
    }

    public void set_serial_number (String serial_number)
    {
        this.serial_number = serial_number;
    }

    public String get_file ()
    {
        return file;
    }

    public void set_file (String file)
    {
        this.file = file;
    }
}
